File: problems/leetcode/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0, max_len = 0;
        unordered_set<char> chars;
        for (int i = 0; i < s.length(); i++) {
            while (chars.find(s[i]) != chars.end()) {
                chars.erase(s[left]);
                left++;
            }
            chars.insert(s[i]);
            max_len = max(max_len, i - left + 1);
        }
        return max_len;
    }    
};

File: problems/leetcode/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = 0
        foundchars = set()
        leftpoint = 0 
        
        for i in range(len(s)):
            while s[i] in foundchars:
                foundchars.remove(s[leftpoint])
                leftpoint +=1
            foundchars.add(s[i])
            longest = max(longest, i - leftpoint + 1)
        return longest